core-types/hash_test.rb

```diff
@@ -59,64 +112,171 @@ def test_member
+    maybe boolean
+    boolean = {"cake": "yummy"}.key?("yummy")
+    maybe boolean
+  end
 
   def test_invert
     skip
     h1 = {"a" => "apple", "b" => "banana"}.invert
-    expected_h1 = __
-    assert_equal expected_h1, h1
-
     h2 = {1 => "one", 2 => "two"}.invert
+    expected_h1 = __
     expected_h2 = __
+    
+    assert_equal expected_h1, h1
     assert_equal expected_h2, h2
   end
 
+  def test_invert_with_symbols
+    skip
+    food_colors = {beets: "purple", kiwis: "green"}.invert
+    foods_with_many_colors = {
+      apples: ["green", "red", "yellow"], 
+      potatoes: ["brown", "red", "purple"]
+    }.invert
+    expected_food_colors = __
+    expected_foods_with_many_colors = __
+    
+    assert_equal expected_food_colors, food_colors
+    assert_equal expected_foods_with_many_colors, foods_with_many_colors
+  end
+
   def test_length
     skip
     assert_equal __, {}.length
     assert_equal __, {"a" => "apple"}.length
-    assert_equal __, {"a" => "apple", "b" => "banana"}.length
+    assert_equal __, {"a" => "apple", "b" => "banana", "c" => "cantaloupe"}.length
   end
 
   def test_merge
     skip
     h1 = {}.merge({})
-    h2 = {}.merge({"a" => "apple"})
-    h3 = {"a" => "apple"}.merge({"b" => "banana"})
-    h4 = {"a" => "apple"}.merge({"a" => "avocado", "b" => "banana"})
+    apple = {"a" => "apple"}
+    h2 = {}.merge(apple)
+    h3 = apple.merge({"b" => "banana"})
+    h4 = apple.merge({"a" => "avocado", "b" => "banana"})
     h5 = {"a" => "apple", "b" => "banana"}.merge({"b" => "breadfruit"})
-
     expected_h1 = __
     expected_h2 = __
+    expected_apple = __
     expected_h3 = __
     expected_h4 = __
     expected_h5 = __
 
     assert_equal expected_h1, h1
     assert_equal expected_h2, h2
+    assert_equal expected_apple, apple
     assert_equal expected_h3, h3
     assert_equal expected_h4, h4
     assert_equal expected_h5, h5
   end
 
+  def test_merge_with_symbols
+    skip
+    new_restaurants = {}.merge({
+      cultura: "Latin American", 
+      wallys: "Wisconsin treats"
+      })
+    name_lengths = {maeby: 5}.merge({selena: 6})
+    saturn = {saturn: 7}
+    planet_rings = saturn.merge({jupiter: 3, uranus: 13, neptune: 6})
+    tv_characters = {
+        battlestar: ["Starbuck", "Apollo", "Boomer"],
+        futurama: ["Fry", "Leela", "Bender", "Zoidberg"]
+      }.merge({breaking_bad: "Walter White"})
+    shape_sides = {circle: 1}.merge({square: 4, circle: "infinite"})
+    expected_new_restaurants = __
+    expected_name_lengths = __
+    expected_planet_rings = __
+    expected_saturn = __
+    expected_tv_characters = __
+    expected_shape_sides = __
+
+    assert_equal expected_new_restaurants, new_restaurants
+    assert_equal expected_name_lengths, name_lengths
+    assert_equal expected_planet_rings, planet_rings
+    assert_equal expected_saturn, saturn
+    assert_equal expected_tv_characters, tv_characters
+    assert_equal expected_shape_sides, shape_sides
+  end
```

Comment:
This one is a bit difficult to read. Did you want to call the method on the hash itself for learning purposes? If not maybe set the hashes to variables at the top and then call the methods on those variables?  Also the indentation is inconsistent between lines 203/204 and 210/211/212.

Reply:
I agree; I was trying to make it consistent with the one above, but wanted to be more explicit with naming, so it got a bit longer. I'll update with your suggestions.